### PR TITLE
51 load subfolders

### PR DESCRIPTION
**Pull request by alemagio**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by alemagio** [comment]
As I said in the issue, still working on it but I think it could be useful to collect some feedback in the meantime.

File: index.js [hunk 1]
```diff
@@ -26,6 +26,68 @@ module.exports = function (fastify, opts, next) {
     return err
   }
 
+  function loadDirs (dirs, cb, plugins, prefix) {
```
Comment (Eomm) [inline index.js:29]: I would put the `cb` at the end

File: index.js [hunk 2]
```diff
@@ -26,6 +26,68 @@ module.exports = function (fastify, opts, next) {
...
+              }
+            }
+          } else {
+            plugins.push({
+              // skip directories without script files inside
+              skip: files.every(name => !scriptPattern.test(name)),
+              file: toLoad
+            })
+          }
+          loadDirs(dirs, cb, plugins, prefix)
```
Comment (Eomm) [inline index.js:78]: Just to share: I'm asking myself if this should be an opt-in option or just "load all the folders".

Right now I think it is fine 👍

Comment (alemagio) [reply index.js:78]: reading the issue comments my idea was:
add an option called `recursive` with boolean value, if `true` it loads recursively all dirs

to customize the folder loading the user can add an `autoload.json` file with some options.

Is that correct?

Comment (Eomm) [reply index.js:78]: I like the 'recursive' flag

I would avoid the autoload.json config.
I think, for example an option sync function filter(path) {} where the user return true or false if he wants to load that folder for example

But I would go through steps 👍

**Comment by mcdonnelldean** [comment]
Can we add a test to ensure this works with the auto-routing functionality already available too.

**Comment by alemagio** [comment]
Added a small fix since I noticed that it did not apply the right prefix in case of multiple subfolders.
Added a test for that scenario and added a test for autoroute @mcdonnelldean , please let me know if it is what you meant.

**Comment by alemagio** [comment]
Is there something i did wrong that broke CI?

**Comment by mcollina** [comment]
I'm not sure!

I'm actually close to land https://github.com/fastify/fastify-autoload/pull/76 which is a significant refactor to simplify esm and ts support. You might want to target that.

**Comment by alemagio** [comment]
Sure

**Comment by mcollina** [comment]
This was implemented as part of #76.

**Pull request closed without merging**